test/unit/gateways/beanstream_ipp_test.rb

```diff
@@ -0,0 +1,213 @@
+      assert_match(%r{<Receipt>receipt<}, data)
+      assert_match(%r{<Amount>100<}, data)
+    end.respond_with(successful_refund_response)
+
+    assert_success response
+  end
+
+  def test_scrub
+    assert @gateway.supports_scrubbing?
+    assert_equal @gateway.scrub(pre_scrubbed), post_scrubbed
+  end
+
+  private
+
+  def pre_scrubbed
+    <<-'PRE_SCRUBBED'
+opening connection to demo.ippayments.com.au:443...
+opened
+starting SSL for demo.ippayments.com.au:443...
+SSL established
+<- "POST /interface/api/dts.asmx HTTP/1.1\r\nContent-Type: text/xml; charset=utf-8\r\nSoapaction: http://www.ippayments.com.au/interface/api/dts/SubmitSinglePayment\r\nAccept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3\r\nAccept: */*\r\nUser-Agent: Ruby\r\nConnection: close\r\nHost: demo.ippayments.com.au\r\nContent-Length: 822\r\n\r\n"
+<- "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<soap:Envelope xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\" xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n  <soap:Body>\n    <SubmitSinglePayment xmlns=\"http://www.ippayments.com.au/interface/api/dts\">\n      <trnXML>\n        <![CDATA[<Transaction>\n  <CustRef>1</CustRef>\n  <Amount/>\n  <TrnType>1</TrnType>\n  <CreditCard Registered=\"False\">\n    <CardNumber>4005550000000001</CardNumber>\n    <ExpM>09</ExpM>\n    <ExpY>2015</ExpY>\n    <CVN>123</CVN>\n    <CardHolderName>Longbob Longsen</CardHolderName>\n  </CreditCard>\n  <Security>\n    <UserName>nmi.api</UserName>\n    <Password>qwerty123</Password>\n  </Security>\n  <TrnSource/>\n</Transaction>\n]]>\n      </trnXML>\n    </SubmitSinglePayment>\n  </soap:Body>\n</soap:Envelope>\n"
+-> "HTTP/1.1 200 OK\r\n"
+-> "Server: Microsoft-IIS/6.0\r\n"
+-> "X-Robots-Tag: noindex\r\n"
+-> "X-Powered-By: ASP.NET\r\n"
+-> "Cache-Control: private, max-age=0\r\n"
+-> "Content-Type: text/xml; charset=utf-8\r\n"
+-> "Content-Length: 767\r\n"
+-> "Date: Fri, 19 Dec 2014 19:55:13 GMT\r\n"
+-> "Connection: close\r\n"
+-> "\r\n"
+reading 767 bytes...
+-> "<?xml version=\"1.0\" encoding=\"utf-8\"?><soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns:xsd=\"http://www.w3.org/2001/XMLSchema\"><soap:Body><SubmitSinglePaymentResponse xmlns=\"http://www.ippayments.com.au/interface/api/dts\"><SubmitSinglePaymentResult>&lt;Response&gt;\r\n\t&lt;ResponseCode&gt;1&lt;/ResponseCode&gt;\r\n\t&lt;Timestamp&gt;20-Dec-2014 06:55:17&lt;/Timestamp&gt;\r\n\t&lt;Receipt&gt;&lt;/Receipt&gt;\r\n\t&lt;SettlementDate&gt;&lt;/SettlementDate&gt;\r\n\t&lt;DeclinedCode&gt;183&lt;/DeclinedCode&gt;\r\n\t&lt;DeclinedMessage&gt;Exception parsing transaction XML&lt;/DeclinedMessage&gt;\r\n&lt;/Response&gt;\r\n</SubmitSinglePaymentResult></SubmitSinglePaymentResponse></soap:Body></soap:Envelope>"
+read 767 bytes
+Conn close
+    PRE_SCRUBBED
+  end
+
+  def post_scrubbed
```

Comment:
could we use the VCR gem to record these responses? It would be easier to re-run in the future

Reply:
If we can leave this feature in future improvement?